### PR TITLE
release switch norm layer

### PR DESCRIPTION
**Pull request by zsdonghao**
### Checklist
- [x] I've tested that my changes are compatible with the latest version of Tensorflow.
- [x] I've read the [Contribution Guidelines](https://github.com/tensorlayer/tensorlayer/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
Switch Normalisation

### Reference
https://zhuanlan.zhihu.com/p/39296570?utm_source=wechat_session&utm_medium=social&utm_oi=984862267107651584

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -99,6 +99,7 @@ To release a new version, please update the changelog as followed:
   - `AMSGrad` added on Optimizers page added (PR #636)
   - Update input scale of VGG16 and VGG19 to 0~1 (PR #736)
 - Layer:
+  - Relase SwitchNormLayer (PR #737)
```
Comment (DEKHTIARJonathan) [inline CHANGELOG.md]: typo error

File: tensorlayer/layers/normalization.py [hunk 2]
```diff
@@ -299,3 +294,77 @@ def __init__(
...
+            epsilon = 1e-5
+
+            batch_mean, batch_var = tf.nn.moments(x, [0, 1, 2], keep_dims=True)
+            ins_mean, ins_var = tf.nn.moments(x, [1, 2], keep_dims=True)
+            layer_mean, layer_var = tf.nn.moments(x, [1, 2, 3], keep_dims=True)
+
+            gamma = tf.get_variable("gamma", [ch], initializer=gamma_init)
+            beta = tf.get_variable("beta", [ch], initializer=beta_init)
+
+            mean_weight = tf.nn.softmax(tf.get_variable("mean_weight", [3], initializer=tf.constant_initializer(1.0)))
```
Comment (DEKHTIARJonathan) [inline tensorlayer/layers/normalization.py]: Are you sure that we should declare a variable and apply a function on it on the same line ?
This will point the variable list of the layer to the wrong tensor (the softmax one, not the mean_weight var).

File: tensorlayer/layers/normalization.py [hunk 3]
```diff
@@ -299,3 +294,77 @@ def __init__(
...
+            var_wegiht = tf.nn.softmax(tf.get_variable("var_weight", [3], initializer=tf.constant_initializer(1.0)))
```
Comment (DEKHTIARJonathan) [inline tensorlayer/layers/normalization.py]: Same remark

File: CHANGELOG.md (hunk 1 above)
Comment (zsdonghao) [reply CHANGELOG.md]: fixed

**Comment by DEKHTIARJonathan** [comment]
@zsdonghao I applied small changes to insure we have the correct tensors in the params attribute ;)

File: tensorlayer/layers/normalization.py [hunk 4]
```diff
@@ -299,3 +300,79 @@ def __init__(
...
+            epsilon = 1e-5
+
+            batch_mean, batch_var = tf.nn.moments(x, [0, 1, 2], keep_dims=True)
+            ins_mean, ins_var = tf.nn.moments(x, [1, 2], keep_dims=True)
+            layer_mean, layer_var = tf.nn.moments(x, [1, 2, 3], keep_dims=True)
+
+            gamma = tf.get_variable("gamma", [ch], initializer=gamma_init)
+            beta = tf.get_variable("beta", [ch], initializer=beta_init)
+
+            mean_weight_var = tf.get_variable("mean_weight", [3], initializer=tf.constant_initializer(1.0))
```
Comment (DEKHTIARJonathan) [inline tensorlayer/layers/normalization.py]: @zsdonghao are you sure we should initialize to 1.0 ?
Due to softmax applied it would seem smarter to initialize at 0.0 to me ... Maybe I didn't understand something ;)

File: tensorlayer/layers/normalization.py [hunk 5]
```diff
@@ -299,3 +300,79 @@ def __init__(
...
+            var_wegiht_var = tf.get_variable("var_weight", [3], initializer=tf.constant_initializer(1.0))
```
Comment (DEKHTIARJonathan) [inline tensorlayer/layers/normalization.py]: @zsdonghao are you sure we should initialize to 1.0 ?
Due to softmax applied it would seem smarter to initialize at 0.0 to me ... Maybe I didn't understand something ;)

Comment (zsdonghao) [reply tensorlayer/layers/normalization.py]: hi, it is correct, after sotfmax, they are averaged. It is not good to initialise them by zero, otherwise, we can't compute the softmax.

**Comment by DEKHTIARJonathan** [comment]
@zsdonghao thanks for your answer ;) Last questions for you, BN implements the `is_train` bool. Do you need a similar behaviour that only trains the mean & var during training ?

**Comment by zsdonghao** [comment]
@DEKHTIARJonathan hi, switchable normalisation does not require separating training and testing.
